### PR TITLE
New link mvtstock to project

### PR DESCRIPTION
**Pull request by aconits**
# New link movement stock to project
Can select project on correction stock view to link the movement then the information can be viewed in "Overview" tab of the project

### CODE REVIEW

File: htdocs/product/class/product.class.php [hunk 1]
```diff
@@ -3518,20 +3518,42 @@ function getLibFinished()
...
 	 * 	@return     int     				<0 if KO, >0 if OK
 	 */
-	function correct_stock($user, $id_entrepot, $nbpiece, $movement, $label='', $price=0, $inventorycode='')
+	function correct_stock($user, $id_entrepot, $nbpiece, $movement, $label='', $price=0, $inventorycode='', $origin_element='', $origin_id=0)
 	{
 		if ($id_entrepot)
 		{
 			$this->db->begin();
 
+			if ($movement == 1)
```
Comment (eldy) [inline htdocs/product/class/product.class.php:3531]: When a stock is decreased a PMP must never be modified (price must be undefined for this), so this line should not be here.

File: htdocs/core/class/html.formprojet.class.php [hunk 2]
```diff
@@ -155,10 +155,8 @@ function select_projects_list($socid=-1, $selected='', $htmlname='projectid', $m
 		if ($socid == 0) $sql.= " AND (p.fk_soc=0 OR p.fk_soc IS NULL)";
 		if ($socid > 0 && empty($conf->global->PROJECT_ALLOW_TO_LINK_FROM_OTHER_COMPANY))  $sql.= " AND (p.fk_soc=".$socid." OR p.fk_soc IS NULL)";
 		if (!empty($filterkey)) {
```
Comment (eldy) [inline htdocs/core/class/html.formprojet.class.php]: It seems a previous fix is lost here.

File: htdocs/core/class/html.formprojet.class.php [hunk 3]
```diff
@@ -155,10 +155,8 @@ function select_projects_list($socid=-1, $selected='', $htmlname='projectid', $m
...
-			$sql .= ' AND (';
-			$sql .= ' p.title LIKE "%'.$this->db->escape($filterkey).'%"';
-			$sql .= ' OR p.ref LIKE "%'.$this->db->escape($filterkey).'%"';
-			$sql .= ')';
+			$sql .= " AND p.title LIKE '%".$this->db->escape($filterkey)."%'";
```
Comment (eldy) [inline htdocs/core/class/html.formprojet.class.php:158]: It seems a previous fix was lost here

File: htdocs/product/class/product.class.php (hunk 1 above)
Comment (aconits) [reply htdocs/product/class/product.class.php:3531]: The PMP is only calculated on increase stock
When we do a negative movement the column price in the database isn't evaluate and I need to save the actual PMP to know which price it cost when is it use this price to show on project tab

**Comment by aconits** [comment]
I close this PR and I'll do it again from clean develop branch because this one come from a 4.0 branch at the beginning

**Pull request closed without merging**